sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -708,17 +732,18 @@ object DataSource extends Logging {
   def lookupDataSourceV2(provider: String, conf: SQLConf): Option[TableProvider] = {
```

Comment:
how do you think of my idea to only put python data source handing in this method? https://github.com/apache/spark/pull/44269/files#diff-2a3ed194aac77f3de25418a74a756d8d821feb2b3d38f4fec144f312e022801aR709

Reply:
yeah I like that idea. Can I do it in a followup though? I would like to extract some changes from your PR, and make another PR.

Reply:
It's not a followup... I have a concern about changing `lookupDataSource` which is only used for the DS v1 path. Let's avoid the risk of breaking anything. It's also less code change if we only instantiate the `PythonTableProvider` here, so that the existing caller of `lookupDataSource` can still instantiate the objects directly instead of calling the new `newDataSourceInstance` function.

Reply:
oh okie dokie. I was actually thinking about porting more changes in your PR. I will fix that one alone here for now.